cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -673,8 +759,8 @@ public String getOutputFormatClassName() {
   @Override
   public void onSuccess() throws DataSetException {
     String outputPath = FileSetArguments.getOutputPath(runtimeArguments);
-    // if there is no output path, the batch job probably would have failed
-    // we definitely can't do much here if we don't know the output path
+    // If there is no output path, it is either using DynamicPartitioner or the job would have failed.
+    // Either way, we can't do much here.
```

Comment:
Yea, seems like we're delegating to the FileSet for all other operations, so makes sense to also call its onSuccess here.